src/control/tools/BaseStrokeHandler.h

```diff
@@ -59,8 +61,11 @@ class BaseStrokeHandler: public InputHandler {
 protected:
     DocumentView view;
     Point currPoint;
-    Point buttonDownPoint;  // used for tapSelect and filtering - never snapped to grid. See startPoint defined in
-                            // derived classes such as CircleHandler.
+    Point buttonDownPoint;  // used for tapSelect and filtering - never snapped to grid.
+    Point startPoint;       // May be snapped to grid
+
+    std::mutex strokeMutex;  // protects InputHandler::stroke
```

Comment:
Why is this mutex not located next to the stroke?

Reply:
Because BaseStrokeHandler::draw uses the stroke, so the stroke needs protecting in this derived class (of InputHandler). On the other hand StrokeHandler::draw uses a mask, so the stroke does not need protecting in this derived class.
I don't know how to make this cleanly...